Functions/xASL_quant_SliceTimeVector.m

```diff
@@ -0,0 +1,75 @@
+function SliceTime = xASL_quant_SliceTimeVector(x,nSlices)
+%xASL_quant_SliceTimeVector Calculates the vector of slice-times for the given number of slices
+%
+% FORMAT: SliceTime = xASL_quant_SliceTimeVector(x,nSlices)
+%
+% INPUT:
+%   x          - struct containing pipeline environment parameters (REQUIRED)
+%   nSlices    - number of slices in ASL native space to calculate the SliceTime for (REQUIRED)
+% OUTPUT:
+%   SliceTime  - a vector of SliceTimes with the length nSlices
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function takes the x.SliceReadoutTime, which can be a vector or scalar and creates a vector
+%              out of it with the correct length. It also checks the x.readout_dim for which it returns 0.
+%
+% 0. Admin
+% 1. ShortestTR
+% 2. Assign the vector value
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE: SliceTime = xASL_quant_SliceTimeVector(x,32)
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+%% -----------------------------------------------------------------------------------------------------------------------------------------------------
+%% 0.Admin
+if nargin < 1 || isempty(x)
+	error('The x-structure needs to be provided');
+end
+
+if nargin < 2 || isempty(nSlices) 
+	error('The nSlices parameter needs to be provided');
+end
+
+% The readout_dim needs to be provided
+if ~isfield(x,'readout_dim')
+	error('x.readout_dim field is missing');
+end
+
+if strcmpi(x.readout_dim, '3D')
+	% For 3D sequences, zero is returned
+	SliceTime = 0;
+	return;
+end
+
```

Comment:
Why do we check a lot of x fields here that are not used below (unless except for the shortest_TR function)?
Doesn't it make more sense to keep this function "x-free"? Or should it also output x?

Reply:
1) We cannot make it x-free. We would have to provide the x.readout_dim and other parameters as parameters, and we would have to check those before calling this function. This was the exact behavior you wanted to avoid, so we hid all these inside.
2) Not true that we check for unnecessary x-fields. readout_dim is needed to return 0 for 3D. SliceReadoutTime is always needed. Nothing else is checked.
3) I wouldn't output x. Because we input SliceReadoutTime (legacy field) and output SliceTime (BIDS field). These will change once we turn to full BIDS later - then we can redo all these everywhere in the code. But to make things easier and more transparent now, I have the legacy input and BIDS output (that way the xASL functions used after this function are already in turned to BIDS compliant) - but I didn't want to mix the inside x-struct now.

Hope this answers all questions ;)